code/modules/medical/diseases/headspider.dm

```diff
@@ -99,40 +98,35 @@
 			if(probmult(40))
-				//ALL of this is awful why are we passing a fuckingf mind in as an arg why does the headspider disease not have a reference to the HEADSPIDER my GOD
-				if(changeto)
+				if(!D.source.changeling)
+					//if the headspider doesn't have a changeling, we create one
+					D.source.mind.add_antagonist(ROLE_CHANGELING, TRUE, FALSE, FALSE, TRUE, ANTAGONIST_SOURCE_SUMMONED, FALSE, FALSE, FALSE)
+					var/datum/antagonist/changeling/antag_datum = D.source.mind.get_antagonist(ROLE_CHANGELING)
+					D.source.changeling = antag_datum.ability_holder
+					logTheThing(LOG_COMBAT, D.source.mind, "became a changeling by infecting [affected_mob] as [D.source].")
+				// Absorb their DNA. Copies identities and DNA points automatically if victim was another changeling. This also inserts them into the hivemind.
+				D.source.changeling.addDna(affected_mob, TRUE)
+				// Remove changeling AH (if any) and copy our own.
+				if (ischangeling(affected_mob))
+					D.source.show_text("[affected_mob] was a changeling! We have incorporated their entire genetic structure.", "blue")
+					affected_mob.remove_ability_holder(/datum/abilityHolder/changeling)
 
-					// Absorb their DNA. Copies identities and DNA points automatically if victim was another changeling. This also inserts them into the hivemind.
-					changeling.addDna(affected_mob, 1)
+				affected_mob.add_existing_ability_holder(D.source.changeling)
+				D.source.changeling.reassign_hivemind_target_mob()
 
-					// Remove changeling AH (if any) and copy our own.
-					if (ischangeling(affected_mob))
-						if (changeto.current)
-							changeto.current.show_text("[affected_mob] was a changeling! We have incorporated their entire genetic structure.", "blue")
-						affected_mob.remove_ability_holder(/datum/abilityHolder/changeling)
-					/*
-					else
-						var/datum/mind/new_ling = affected_mob.mind
-						ticker.mode.traitors += new_ling
-						new_ling.special_role = "changeling"
-					*/
+				// Transfer player control.
+				D.source.changeling = null //so the spider doesn't have a ref to our holder as well
+				D.source.mind.transfer_to(affected_mob)
+				affected_mob.change_misstep_chance(-INFINITY)
+				affected_mob.show_text("<h3>We have assumed control of the new host.</h3>", "blue")
+				logTheThing(LOG_COMBAT, affected_mob, "'s headspider successfully assumes control of new host at [log_loc(affected_mob)].")
 
-					affected_mob.add_existing_ability_holder(changeling)
-					changeling.reassign_hivemind_target_mob()
+				D.stealth_asymptomatic = TRUE //Retain the disease but don't actually do anything with it
+				//kill the headspider, so if something causes it to drop it doesn't look alive with no mind
+				D.source?.death(FALSE)
+				SPAWN(2 MINUTES) //Disease stays for two minutes after a complete infection, then it removes itself.
+					//the headspider gets fully absorbed
+					D.source?.set_loc(null)
+					qdel(D.source)
```

Comment:
```suggestion
					qdel(D.source)
					D.source = null
```
Idk if /datum/ailment/parasite/headspider GC's correctly at the moment or not, but maybe for GC? Alternatively, since cure_disease_by_path can be called on it, maybe would be better in disposing()